### DISCUSSION TITLE
How to set Problems View filter from extension

### DISCUSSION THREAD

**Discussion by vaclavHala** [microsoft/vscode-discussions#1432] [Extension Development QnA] [answered]
I have an extension which has a notion of projects that can be located anywhere inside the workspace. The extension emits problem markers (`vscode.Diagnostic`) for various resources inside the projects. I'd like to provide a convenience command to the user which would show problems for selected project. 

To do this (I think) I need to simply focus the Problems view (I know how to do this) and then somehow set the filter to `${projectRoot}/*` (this I did not figure out how to do). Is this approach sensible or is there different better way? 

I have found there exists [this method](https://github.com/microsoft/vscode/blob/ba3cdbc4b3e2a8146d77fb2c796cc92e132a9970/src/vs/workbench/browser/parts/views/viewFilter.ts#L135) to set the filter, but I see no way to get to it from my extension.

**Comment by robconery** [comment]
This might be a better question for [StackOverflow](https://stackoverflow.com/questions/tagged/vscode-extensions) if it's "is this a good way to do this thing". If you have a more specific question about developing an extension we can do our best to help.

**Reply by vaclavHala** [reply]
Hi, I'm not asking if this is a good approach or not, I say in the original question what I did not figure out how to do at all ("somehow set the filter to `${projectRoot}/` "). 

The "Is this approach sensible or is there different better way?" is merely asking if I was even looking in the right place (for the thing I did not find...) or if there is another approach I did not think about at all (so I don't need the thing I did not find).

So the specific question is - Can I somehow force text into the `filter` of problems view from code of my extension? If not, is there some other way I can use to control what the problems view shows - filter its content somehow on behalf of the user - from code of my extension?

**Comment by isidorn** [comment, marked as answer]
Not possible from an extension. Sorry!
